qiskit_ibm_provider/job/ibm_circuit_job.py

```diff
@@ -880,124 +755,25 @@ def _format_message_from_error(self, error: Dict) -> str:
-                last_data = (status, queue_info)
-            logger.debug(
-                "Invoking callback function, job status=%s, queue_info=%s",
-                status,
-                queue_info,
-            )
-            callback(self.job_id(), status, self, queue_info=queue_info)
-
-    def _get_status_position(
-        self, api_status: str, api_info_queue: Optional[Dict] = None
-    ) -> Tuple[JobStatus, Optional[QueueInfo]]:
-        """Return the corresponding job status for the input server job status.
-
-        Args:
-            api_status: Server job status
-            api_info_queue: Job queue information from the server response.
-
-        Returns:
-            A tuple of job status and queue information (``None`` if not available).
-
-        Raises:
-             IBMJobApiError: if unexpected return value received from the server.
-        """
-        queue_info = None
-        status = api_status_to_job_status(api_status)
-        if api_status == ApiJobStatus.QUEUED.value and api_info_queue:
-            queue_info = QueueInfo(job_id=self.job_id(), **api_info_queue)
-
-        if status is not JobStatus.QUEUED:
-            queue_info = None
-
-        return status, queue_info
-
-    def _get_qobj(self) -> Optional[Union[QasmQobj, PulseQobj]]:
-        """Return the Qobj for this job.
-
-        Returns:
-            The Qobj for this job, or ``None`` if the job does not have a Qobj.
-
-        Raises:
-            IBMJobApiError: If an unexpected error occurred when retrieving
-                job information from the server.
-        """
-        if not self._kind:
-            return None
-
-        if not self._qobj:
-            with api_to_job_error():
-                qobj = self._api_client.job_download_qobj(
-                    self.job_id(), self._use_object_storage
-                )
-                self._qobj = dict_to_qobj(qobj)
-
-        return self._qobj
-
-    def _extract_client_version(self, data: Dict[str, str]) -> Dict:
+    def _extract_client_version(self, data: str) -> Dict:
         """Extract client version from API.
 
         Args:
             data: API client version.
 
         Returns:
             Extracted client version.
-        """
-        if data:
-            if data.get("name", "").startswith("qiskit"):
-                return dict(zip(data["name"].split(","), data["version"].split(",")))
 
-            return {data.get("name", "unknown"): data.get("version", "unknown")}
+        Additional info:
+            The runtime client returns the version as a string, e.g.
+            "0.1.0,0.21.2,0.10.4"
+            Where the numbers represent versions of qiskit-ibm-provider, qiskit-terra and qiskit-aer
+        """
+        if data is not None:
+            if "," not in data:  # sometimes only the metapackage version is returned
+                return {"qiskit": data}
+            client_components = ["qiskit-ibm-provider", "qiskit-terra", "qiskit-aer"]
```

Comment:
```suggestion
            client_components = ["qiskit-ibm-provider", "qiskit-terra"]
```

Let's return just the provider and terra version, since aer is an optional package, and thus the third value may be a different optional package.